### ISSUE TITLE
Dynamically update rate limits

### ISSUE THREAD

**Issue by Edwin9292** [MarcGiffing/bucket4j-spring-boot-starter#202]
Hello, 

For my Software Engineering Thesis I was looking for a rate limiting solution that integrates with a Spring environment and found your library. Your solution works for almost all requirements that I had, except dynamically updating rate limits in a distributed environment. 

After discussing this with my employer we decided to fork your library and implement the logic as part of my thesis. My goal is to contribute my changes to your library, but I understand that this willl be a large change. Is there any way i could contact you to further discuss about this? 

Greetings,

**Comment by vladimir-bukhtoyarov** [comment]
@Edwin9292  hello,

bucket4j-spring-boot-starter is the starter for [Bucket4j](https://github.com/bucket4j/bucket4j)
Bucket4j by itself already supports the both config changing style:
* [Explicit configuration replacement](https://bucket4j.com/8.7.0/toc.html#configuration-replacement) when config change is triggered by something outside application.
* [Implicit configuration replacement](https://bucket4j.com/8.7.0/toc.html#implicit-configuration-replacement) when config change is triggered vy changing in config files.

I do not think that creation of independent fork of bucket4j-spring-boot-starter is right idea. I suppose that pull request will be fully  enough, you need just to add to existed starter the functionality that already covered in underlyed library.

P.S. I am not maintainer of bucket4j-spring-boot-starter, I am maintainer of bucket4j.

**Comment by Edwin9292** [comment]
@vladimir-bukhtoyarov 
Thank you for your response. 

I know that Bucket4J offers this functionality. For my solution I in fact use the implicit replace :). 
The reason I created a fork is because I don't know if the maintainer of this library is interested in such functionality, but I will implement it for my thesis anyways. That's also why I wrote this issue, to ask if the maintainer would be interested in accepting such a feature (with a pull request ofcourse) once it's finished.
